WcaOnRails/app/models/competition.rb

```diff
@@ -824,6 +824,25 @@ def to_wcif
     }
   end
 
+  def load_wcif!(wcif)
+    # Remove extra events.
+    self.competition_events.each do |competition_event|
+      competition_event.destroy! unless wcif["events"].find { |wcif_event| wcif_event["id"] == competition_event.event.id }
+    end
+
+    # Create missing events.
+    wcif["events"].each do |wcif_event|
+      competition_events.find_or_create_by!(event_id: wcif_event["id"])
+    end
+
+    # Update all events.
+    wcif["events"].each do |wcif_event|
+      competition_events.find_by_event_id!(wcif_event["id"]).load_wcif!(wcif_event)
+    end
```

Comment:
Oh, I actually intentionally wrote this in 3 steps. See my later comment in the competition_event class for why. It's probably safe to change this code block as you suggest, but I feel more comfortable with the symmetry with the `competition_event.load_wcif!` method.